### PR TITLE
netavark: add macvlan driver support

### PR DESCRIPTION
**Pull request by Luap99**
#### What this PR does / why we need it:

Netvark supports macvlan networks now so podman should be able to create
them. Unfortunately netavark does not support DHCP yet so we have force
that the user specifies a subnet.

### CODE REVIEW

File: libpod/network/netavark/config.go [hunk 1]
```diff
@@ -102,7 +103,36 @@ func (n *netavarkNetwork) networkCreate(newNetwork types.Network, defaultNet boo
...
+		if newNetwork.Internal {
+			return nil, errors.New("internal is not supported with macvlan")
+		}
+		if newNetwork.NetworkInterface != "" {
+			interfaceNames, err := internalutil.GetLiveNetworkNames()
+			if err != nil {
+				return nil, err
+			}
+			if !util.StringInSlice(newNetwork.NetworkInterface, interfaceNames) {
+				return nil, errors.Errorf("parent interface %s does not exists", newNetwork.NetworkInterface)
```
Comment (TomSweeneyRedHat) [inline libpod/network/netavark/config.go:119]:
```suggestion
				return nil, errors.Errorf("parent interface %s does not exist", newNetwork.NetworkInterface)
```

File: libpod/network/netavark/config.go [hunk 2]
```diff
@@ -102,7 +103,36 @@ func (n *netavarkNetwork) networkCreate(newNetwork types.Network, defaultNet boo
...
+			}
+		}
+		if len(newNetwork.Subnets) == 0 {
+			return nil, errors.Errorf("macvlan driver needs at least one subnet specified, DHCP is not supported with netavark")
```
Comment (TomSweeneyRedHat) [inline libpod/network/netavark/config.go:123]: I just wonder if an end-user will be confused by "netavark" in this message. Maybe "Podman's network stack" instead?

Comment (Luap99) [reply libpod/network/netavark/config.go:123]: No netvark will at least bring up search results. "Podman's network stack" can mean anything even CNI.

File: libpod/network/netavark/config.go [hunk 3]
```diff
@@ -102,7 +103,36 @@ func (n *netavarkNetwork) networkCreate(newNetwork types.Network, defaultNet boo
...
+		}
+		newNetwork.IPAMOptions["driver"] = types.HostLocalIPAMDriver
+
+		// validate the given options, we do not need them but just check to make sure they are valid
+		for key, value := range newNetwork.Options {
+			switch key {
+			case "mode":
+				if !util.StringInSlice(value, []string{"", "bridge", "private", "vepa", "passthru"}) {
```
[the remark this answers is not in the document]
Comment (Luap99) [reply libpod/network/netavark/config.go]: fixed

File: libpod/network/netavark/config.go (hunk 2 above)
Comment (rhatdan) [reply libpod/network/netavark/config.go:123]: Is this forever or do we plan on adding dhcp support at some point?

Comment (Luap99) [reply libpod/network/netavark/config.go:123]: we plan to support this but it is not trivial, basically needs another deamon

**Comment by Luap99** [comment]
Moved to c/common, see https://github.com/containers/common/pull/863

**Pull request closed without merging**